### PR TITLE
DOCS-2237: Review SLAM and baseRC snippets

### PR DESCRIPTION
**Pull request by skyleilani**

### CODE REVIEW

File: docs/mobility/base-rc/_index.md [hunk 1]
```diff
@@ -140,10 +140,14 @@ Close out of all remote control related systems.
 For more information, see the [Go SDK Docs](https://pkg.go.dev/go.viam.com/rdk/services/baseremotecontrol).
 
 ```go {class="line-numbers linkable-line-numbers"}
-baseRCService, err := baseremotecontrol.FromRobot(robot, "my_base_rc_service")
+baseRCService, err := baseremotecontrol.FromRobot(machine, "my_base_rc_service")
+if err != nil {
```
Comment (npentrel) [inline docs/mobility/base-rc/_index.md:144]: We haven't done these checks in the rest of the docs. I think best to remove here?

File: docs/mobility/slam/_index.md [hunk 2]
```diff
@@ -104,10 +104,14 @@ pose = await slam.get_position()
...
-slam_svc, err := slam.FromRobot(robot, "my_slam_service")
+mySlam, err := slam.FromRobot(machine, "my_slam_service")
+if err != nil {
+  logger.Error(err)
+  return
+}
 
 // Get the current position of the specified source component in the SLAM map as a Pose.
-pos, name, err := slam_svc.GetPosition(context.Background())
+pos, name, err := mySlam.Position(context.Background())
```
Comment (npentrel) [inline docs/mobility/slam/_index.md:114]: Nice catch

File: docs/mobility/slam/_index.md [hunk 3]
```diff
@@ -143,19 +147,23 @@ pcd_map = await slam_svc.get_point_cloud_map()
 **Parameters:**
 
 - `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
-
-**Returns:**
+- `returnEditedMap` [(bool)](https://pkg.go.dev/builtin#bool): A flag that determines if the method should return the edited version of the point cloud map or the original map. Setting this parameter to `true` triggers a formatted map to be retrieved.
+  **Returns:**
```
Comment (npentrel) [inline docs/mobility/slam/_index.md:151]:
```suggestion

**Returns:**
```

File: docs/mobility/base-rc/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/mobility/base-rc/_index.md:144]: Oh interesting. Yeah alright then let's keep them. We could also consider adding error checking everywhere 🤷‍♀️

Comment (npentrel) [reply docs/mobility/base-rc/_index.md:144]: Perfect

**Review by npentrel — APPROVED** [review]

File: docs/components/generic/_index.md
Comment (JessamyT) [inline docs/components/generic/_index.md:1]: It seems that this PR should update the generic service, not the generic component. 2231 looks like it's for the generic component.

Comment (npentrel) [reply docs/components/generic/_index.md:1]: I think the code is correct though, so it's a bit of a drive by fix?

**Pull request merged**